wire-ios-data-model/Source/Model/Message/ConversationMessage.swift

```diff
@@ -103,6 +103,8 @@ public protocol ZMConversationMessage: NSObjectProtocol {
     var locationMessageData: LocationMessageData? { get }
 
     var usersReaction: [String: [UserType]] { get }
+    var reactionDates: [String: Date] { get }
+    func reactionsSortedByCreationDate() -> [ReactionData]
```

Comment:
I'm wondering perhaps if it's better to replace `usersReactions: [String: [UserType]]` with something like `reactions: Set<ReactionData>`, then adding any kind of extension helpers to work with that data.

For instance:

```swift
let sortedReactions = message.reactions.sorted(by: \.creationDate)

```

Reply:
Yeah, we can do that

Reply:
However we already have `reactions` field which holds `Reaction` models.

Reply:
Then `reactionData`?